web-client/src/modules/create/components/PostSummary.tsx

```diff
@@ -0,0 +1,85 @@
+import { Button } from 'antd';
+import React, { useState } from 'react';
+import { useHistory } from 'react-router-dom';
+import { MyRequestPostsLocationUrl } from 'src/modules/requests/constants';
+import styled from 'styled-components';
+
+import WebClientMap from '../../../components/WebClientMap/WebClientMap';
+import PostConfirmation from './PostConfirmation';
+
+const PostSummary: React.FC<PostSummaryProps> = ({
+  prevHandler: onPrev,
+  postDetails,
+  postLocation,
+  submitRequest,
+}) => {
+  const { coords } = postLocation;
+  const [showConfirmationPage, setShowConfirmationPage] = useState(false);
+  const history = useHistory();
+
+  return (
+    <>
+      {showConfirmationPage && (
+        <PostConfirmation
+          showModal={showConfirmationPage}
+          closeModal={() => {
+            setShowConfirmationPage(false);
+            // because I could observe race conditions in cloud function
```

Comment:
does this mean that the my request posts page loads before the request is submitted to firebase, and so does not show the new request? I believe you could handle this by moving the history.replace call into the container, and waiting for the request submission to complete before calling it.